extensions/panache/panache-common/runtime/src/main/java/io/quarkus/panache/common/Sort.java

```diff
@@ -41,6 +42,8 @@
     }
 
     public class Column {
+
+        private final Pattern COLUMN_PATTERN = Pattern.compile("^[a-zA-Z_][a-zA-Z0-9_]*$");
```

Comment:
```suggestion
        private static final Pattern COLUMN_PATTERN = Pattern.compile("^[a-zA-Z_][a-zA-Z0-9_]*$");
```

Reply:
Are we sure this is OK and covering all the possibilities? Where does this pattern come from?

Reply:
At the beginning, I was more thinking about escaping the column names. And the hard part is probably to get the dialect at some point to get it to properly escape things.

Reply:
@gsmet, as you said, this pattern may not be valid for all databases. 
Can we do a simple quote escape here ? Getting the dialect here seems complicated.

Reply:
While I agree it's probably be a bit difficult, this can cause all sorts of security issues so I would prefer we get this right.

@Sanne any advice on how we could get Hibernate ORM to properly escape this?

Reply:
This looks suspicious indeed. Why does it need to be validated, couldn't we pass the sort intruction to ORM directly?

Some background: in ORM 5.x there have been various attempts in the Dialets to automagically escape / veto column names, they had all to be rolled back as they would break in various ways - I don't remember the details but yes it's dangeour to impose such a rule without a thoutough analysis.

The proper solution is included in the design of ORM6, so I would suggest to just let the user be responsible for using the right names.

Reply:
The issue is that the sort column is directly appended to the generated HQL so you basically end up with an HQL injection if you're not cautious about it.

It somehow needs to be properly quoted/escaped/whatever makes sense.

Reply:
Ok, but is "HQL injection" actually a dangeous thing? We're not sending it to the DB unparsed.

Reply:
Hmmm, unparsed no, sure. But can't you completely change the meaning of the query just with carefully forged sort column?

Reply:
I suppose this could be explored - never seen it done. Also I just realized that these columns names are appended into HQL, didn't expect that